### PR TITLE
Hapi indexer evm

### PR DESCRIPTION
**Pull request by hlgltvnnk**

### CODE REVIEW

**Review by ars9 — APPROVED** [review]

File: indexer/tests/mocks/evm_mock.rs [hunk 1]
```diff
@@ -1,50 +1,493 @@
...
+    }
+
+    fn entity_getters_mock(&mut self, data: Vec<PushData>) {
+        data.iter().for_each(|data| self.processing_data_mock(data));
     }
 
-    fn get_cursor(_batch: &[TestBatch]) -> IndexingCursor {
-        unimplemented!()
+    fn fetching_jobs_mock(&mut self, batches: &[TestBatch], cursor: &IndexingCursor) {
+        let mut to_block = 0;
```
Comment (ProstoSergei) [inline indexer/tests/mocks/evm_mock.rs:97]: You can put this variable in for loop

File: indexer/tests/mocks/near_mock.rs [hunk 2]
```diff
@@ -65,6 +67,14 @@ impl RpcMock for NearMock {
             .expect("Failed to convert")
     }
 
+    fn generate_address() -> String {
+        format!("{}.near", rand::random::<u16>())
+    }
+
+    fn get_delay_multiplier() -> u32 {
+        17
```
Comment (ProstoSergei) [inline indexer/tests/mocks/near_mock.rs:75]: Why did you change it to 17?

File: indexer/tests/mocks/mod.rs [hunk 3]
```diff
@@ -22,7 +22,11 @@ pub mod near_mock;
 pub mod solana_mock;
 pub mod webhook_mock;
 
+pub const PAGE_SIZE: u64 = 6;
```
Comment (ProstoSergei) [inline indexer/tests/mocks/mod.rs:25]: Can we set here more realistic page size?

File: indexer/src/indexer/client/near.rs
Comment (ProstoSergei) [inline indexer/src/indexer/client/near.rs:1]: After changes here if page_size > latest_block test failed

File: client.rs/tests/near/setup.rs
Comment (ProstoSergei) [inline client.rs/tests/near/setup.rs:1]: after changes here this test doesn't open docker anymore

File: indexer/tests/mocks/near_mock.rs (hunk 2 above)
Comment (hlgltvnnk) [reply indexer/tests/mocks/near_mock.rs:75]: Because this is amount of fetched blocks. Upd: decreased this value to amount of batches + 1

File: indexer/tests/mocks/mod.rs (hunk 3 above)
Comment (hlgltvnnk) [reply indexer/tests/mocks/mod.rs:25]: This number corresponds to one batch of test data - based on this number, the indexer fetches test data, if we make this number more realistic, we will need to increase the amount of test data

File: indexer/tests/configuration.near.toml
[the remark this answers is not in the document]
Comment (hlgltvnnk) [reply indexer/tests/configuration.near.toml:1]: There is a sample configuration + everything described in the readme, why do we need another one in the repo?

[the remark this answers is not in the document]
Comment (ars9) [reply indexer/tests/configuration.near.toml:1]: Don't we use it for tests?

File: indexer/tests/mocks/mod.rs (hunk 3 above)
Comment (ProstoSergei) [reply indexer/tests/mocks/mod.rs:25]: Why "we will need to increase the amount of test data"

Comment (hlgltvnnk) [reply indexer/tests/mocks/mod.rs:25]: Because one of the purposes of the tests is to check fetching with batches, if we increase the page size but leave the same amount of test data, then all the data from one index launch will be included in one batch, which contradicts one of the purposes of the tests

File: indexer/tests/configuration.near.toml
[the remark this answers is not in the document]
Comment (hlgltvnnk) [reply indexer/tests/configuration.near.toml:1]: Nope

File: indexer/src/indexer/client/near.rs [hunk 4]
```diff
@@ -118,7 +118,7 @@ pub(super) async fn fetch_near_jobs(
     if start_block < latest_block {
         tracing::info!(start_block, "Fetching near jobs from");
 
-        let final_block = min(PAGE_SIZE.to_owned() + start_block, latest_block);
+        let final_block = min(PAGE_SIZE.to_owned() - 1 + start_block, latest_block);
```
Comment (ProstoSergei) [inline indexer/src/indexer/client/near.rs:121]: Why this -1 is needed?

Comment (hlgltvnnk) [reply indexer/src/indexer/client/near.rs:121]: Substracting 1 from page size because the result will include filter limits
